python/ray/_raylet.pyx

```diff
@@ -2602,6 +2602,8 @@ def maybe_initialize_job_config():
             log_config_dict = logging_config._get_dict_config()
         if log_config_dict:
             try:
+                from ray._private.ray_logging import setup_log_record_factory
+                setup_log_record_factory()
                 logging.config.dictConfig(log_config_dict)
```

Comment:
Given we now have two configs: log record factory and dictConfig. Let's create a `LoggingConfig._apply()` method and do all the configurations there.

Also I guess we only need to have log record factory if the encoding is JSON.

Reply:
> Given we now have two configs: log record factory and dictConfig. Let's create a LoggingConfig._apply() method and do all the configurations there.

What does `LoggingConfig._apply()` do? Does it (1) call `setup_log_record_factory` (2) call `logging.config.dictConfig`?

Reply:
> Also I guess we only need to have log record factory if the encoding is JSON.

Why?

Reply:
> What does LoggingConfig._apply() do? Does it (1) call setup_log_record_factory (2) call logging.config.dictConfig

Yes. Basically encapsulate all the configurations needed by the LoggingConfig.

Reply:
> Why?

I think @GeneDer probably only cares about it if it's JSON. For TEXT which is local development asctime is enough I think but it's probably no harm to add timestamp_ns in this case as well.

Reply:
Yes, I (and the product) only care about this when using JSON encoding

Reply:
Update https://github.com/ray-project/ray/pull/46074/commits/ea8de996a5bfe5863e7254e2e7f4dd63c73e7e7d